app/components/navBar/navBar.js

```diff
@@ -0,0 +1,207 @@
+/**
+ * Created by gaston on 8/8/16.
+ */
+
+var navInst = undefined;
+
+module.exports = navInst ? navInst : navInst = NavBar();
+
+/**
+ * @returns {NavBar}
+ * @constructor
+ */
+function NavBar() {
+  let self = this;
+  var config;
+
+  Object.assign( self, {
+    addBackAction: addBackAction,
+    addOptionButton: addOptionButton,
+    setTitle: setTitle,
+    hide: hide,
+    show: show,
+    addActionButton: addActions,
+    open: open,
+    reset: resetNavBar
+  } );
+
+  init();
+  return self;
+
+  /**
+   * Init NavBar
+   */
+  function init() {
+
+    let navEle = $( '.jjb-navbar' );
```

Comment:
Rename this to js-jjb-navbar. This is to differentiate a selector for css and selectors for be used by javascript. If you have doubts, we can talk about it.